internal/provider/service_account_resource.go

```diff
@@ -342,17 +358,32 @@ func getNamespaceAccessesFromServiceAccountModel(ctx context.Context, diags diag
 
 func updateServiceAccountModelFromSpec(ctx context.Context, diags diag.Diagnostics, state *serviceAccountResourceModel, serviceAccount *identityv1.ServiceAccount) {
 	state.ID = types.StringValue(serviceAccount.GetId())
-	state.State = types.StringValue(serviceAccount.GetState())
+	stateStr, err := enums.FromResourceState(serviceAccount.GetState())
+	if err != nil {
+		diags.AddError("Failed to convert resource state", err.Error())
+
+	}
+	state.State = types.StringValue(stateStr)
 	state.Name = types.StringValue(serviceAccount.GetSpec().GetName())
-	state.AccountAccess = internaltypes.CaseInsensitiveString(serviceAccount.GetSpec().GetAccess().GetAccountAccess().GetRole())
+	role, err := enums.FromAccountAccessRole(serviceAccount.GetSpec().GetAccess().GetAccountAccess().GetRole())
+	if err != nil {
+		diags.AddError("Failed to convert account access role", err.Error())
```

Comment:
Now that we are adding an error here i think the caller of `updateServiceAccountModelFromSpec` needs to handle the case where an error is present in `diags` and return early.

Reply:
It might be cleaner to have `updateServiceAccountModelFromSpec` return an error and the callers to do normal error checking on it, letting them inject it into `diags` if necessary. Thoughts?

Reply:
Ack. 
I moved to updateServiceAccountModelFromSpec returning a diagnostics object and checking for errors on the caller side.